src/main/java/lt/vu/trip/controller/TripController.java

```diff
@@ -16,14 +20,59 @@
 	private TripService tripService;
 
 	@GetMapping("")
-	public ResponseEntity<List<Trip>> getAllTrips() {
-		List<Trip> trips = tripService.getAll();
-		return ResponseEntity.ok(trips);
+	public ResponseEntity<TripListResponse> getAllTrips(@Valid TripRequestParams requestParams) {
+		TripSearchCriteria criteria = buildSearchCriteriaFromRequest(requestParams);
+		Page<Trip> trips = tripService.getAll(requestParams.getPage(), requestParams.getResultsPerPage(), criteria);
+		TripListResponse response = buildListResponseFromPage(trips);
+		return ResponseEntity.ok(response);
+	}
+
+	@GetMapping("/my")
+	public ResponseEntity<TripListResponse> getCurrentUserTrips(@Valid TripRequestParams requestParams) {
+		TripSearchCriteria criteria = buildSearchCriteriaFromRequest(requestParams);
+		Page<Trip> trips = tripService.getCurrentUserParticipatingIn(
+				requestParams.getPage(), requestParams.getResultsPerPage(), criteria);
+		TripListResponse response = buildListResponseFromPage(trips);
+		return ResponseEntity.ok(response);
+	}
+
+	@GetMapping("/invitation")
+	public ResponseEntity<TripListResponse> getInvitations(@Valid TripRequestParams requestParams) {
+		TripSearchCriteria criteria = buildSearchCriteriaFromRequest(requestParams);
+		Page<Trip> trips = tripService.getCurrentUserInvitedIn(
+				requestParams.getPage(), requestParams.getResultsPerPage(), criteria);
+		TripListResponse response = buildListResponseFromPage(trips);
+		return ResponseEntity.ok(response);
+	}
+
+	@GetMapping("/organized")
+	public ResponseEntity<TripListResponse> getOrganized(@Valid TripRequestParams requestParams) {
+		TripSearchCriteria criteria = buildSearchCriteriaFromRequest(requestParams);
+		Page<Trip> trips = tripService.getOrganizedByCurrentUser(requestParams.getPage(), requestParams.getResultsPerPage(), criteria);
+		TripListResponse response = buildListResponseFromPage(trips);
+		return ResponseEntity.ok(response);
 	}
 
 	@PostMapping("")
 	public ResponseEntity<Trip> createTrip(@RequestBody Trip trip) {
 		tripService.createNew(trip);
 		return ResponseEntity.ok(trip);
 	}
+
+	private TripSearchCriteria buildSearchCriteriaFromRequest(TripRequestParams requestParams) {
+		return TripSearchCriteria.builder()
+				.startDateFrom(requestParams.getStartDateFrom())
+				.startDateTo(requestParams.getStartDateTo())
+				.endDateFrom((requestParams.getEndDateFrom()))
+				.endDateTo((requestParams.getEndDateTo()))
+				.build();
+	}
+
+	private TripListResponse buildListResponseFromPage(Page<Trip> trips) {
```

Comment:
Is it okay to have such private functions here? As this is about response building, I put it here.

Reply:
Usually the service is response for creating the response so I guess it would be better to move this to the service or have a separate class for this.